cmake/gpu.cmake

```diff
@@ -38,22 +41,45 @@ add_dependencies(gpu webgpulib)
 target_include_directories(gpu INTERFACE ${TARGET_FILE_PATH})
 
 # Add headers webgpu.h
-target_include_directories(wgpu INTERFACE ${TARGET_FILE_PATH}/third_party/headers)
+target_include_directories(wgpu
+                           INTERFACE ${TARGET_FILE_PATH}/third_party/headers)
 if(WIN32)
-    set(DLL_PATH "${TARGET_FILE_PATH}/third_party/lib/libdawn_${ARCH}_${BUILD_TYPE}.dll")
+    set(DLL_PATH
+        "${TARGET_FILE_PATH}/third_party/lib/libdawn_${ARCH}_${BUILD_TYPE}.dll")
     if(EXISTS ${DLL_PATH})
-            file(COPY ${DLL_PATH} DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
-            target_link_libraries(webgpulib INTERFACE ${DLL_PATH})
+        file(COPY ${DLL_PATH} DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+        target_link_libraries(webgpulib INTERFACE ${DLL_PATH})
     else()
-    message(FATAL_ERROR "libdawn dll not found at: ${DLL_PATH}")
-endif()
+        message(FATAL_ERROR "libdawn dll not found at: ${DLL_PATH}")
+    endif()
 else()
-    find_library(LIBDAWN dawn REQUIRED PATHS "${TARGET_FILE_PATH}/third_party/lib")
+    find_library(LIBDAWN dawn PATHS "${TARGET_FILE_PATH}/third_party/lib")
     if(LIBDAWN)
         message(STATUS "Found libdawn: ${LIBDAWN}")
-    # Link against libdawn
+        # Link against libdawn
         target_link_libraries(webgpulib INTERFACE ${LIBDAWN})
+        # if not found, try download from release
     else()
-        message(FATAL_ERROR "libdawn not found")
+        message("libdawn not found, try downloading from the release")
+        if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+            set(libdawn_ext "dylib")
+        elseif(UNIX)
+            set(libdawn_ext "so")
+        endif()
+        FetchContent_Declare(
+            libdawn
+            URL https://github.com/austinvhuang/dawn-artifacts/releases/download/prerelease/libdawn.${libdawn_ext}
```

Comment:
I'm not sure if they're hosted yet but we should use the libdawn_${ARCH}_${BUILD_TYPE} convention unless @austinvhuang wants to migrate away from that.

Reply:
For windows yes but windows libraries are in another place. Under this url it's just libdawn.{so/dylib}. Maybe I can add one for Windows with the architecture and build type for windows.

Reply:
I think we were going to structure all the libs with this sort of convention, could be mistaken, need Austin to clarify. Other OS will have debug/release/architecture differences too, so I think we wanted to capture all with a generalized approach.